Marlin/src/Marlin.cpp

```diff
@@ -508,7 +508,7 @@ void manage_inactivity(const bool ignore_stepper_queue/*=false*/) {
     const int HOME_DEBOUNCE_DELAY = 2500;
     if (!IS_SD_PRINTING() && !READ(HOME_PIN)) {
       if (!homeDebounceCount) {
-        queue.inject_P(PSTR("G28"));
+        queue.enqueue_now_P(PSTR("G28"));
```

Comment:
Seems to me that a `G28` would be fine to have cancel the remainder of the previous injected queue, otherwise who knows what would follow?

Reply:
It is possible for any of the “now” variants to call `idle` which calls into this. So this would need to be made properly re-entrant also. Currently it does not change `homeDebounceCount` before calling the “now” method, and this will likely lead to recursion.

Reply:
This one I didn't necessarily want to keep the now however that was my only other option with Progmem. I considered adding another standard cue function to handle things from progmem but that was more than I was getting into late last night....